cpp/include/cudf/types.h

```diff
@@ -177,7 +177,28 @@ typedef enum {
   N_GDF_AGG_OPS,      ///< The total number of aggregation operations. ALL NEW OPERATIONS SHOULD BE ADDED ABOVE THIS LINE
 } gdf_agg_op;
 
+/**
+ * @brief These enums indicate the supported reduction operations that can be
+ * performed on a column
+ */
+typedef enum {
+  GDF_REDUCTION_SUM = 0,        ///< Computes the sum of all values in the column
+  GDF_REDUCTION_MIN,            ///< Computes the minimum value of all values in the column
+  GDF_REDUCTION_MAX,            ///< Computes the maximum value of all values in the column
+  GDF_REDUCTION_PRODUCTION,     ///< Computes the multiplicative product value of all values in the column
+  GDF_REDUCTION_SUMOFSQUARES,   ///< Computes the sum of squares of the values in the column
+} gdf_reduction_op;
 
+/**
+ * @brief These enums indicate the supported operations of prefix scan that can be
+ * performed on a column
+ */
+typedef enum {
```

Comment:
You could also just use the enum for binary_ops that @devavret added since these are all binary ops, effectively. Not all will apply, but at least we avoid proliferation of enums.

Reply:
Actually I agree to merge .

But which should I merge for enums?
1. create gdf_op enum and share by `reduction` and `scan`
2. add `min`, `max`, `sum_of_squares` into `gdf_binary_operator` and share it by `reduction`, `scan`, `binary_operation`

If we choose #1, I can implement it in this PL.
If we choose #2, I prefer to create an issue and separate PL from this.
Because there is also simliar structure for device operator.
We may share them. How do we think?  @harrism  @jrhemstad  @devavret 

https://github.com/rapidsai/cudf/blob/b63e971f9c2348db6c12882c882139b2e0988d1e/cpp/src/utilities/device_atomics.cuh#L314-L323
https://github.com/rapidsai/cudf/blob/b63e971f9c2348db6c12882c882139b2e0988d1e/cpp/src/binary/binary_ops.cu#L121-L127

Reply:
Thinking about it more... I think either way I prefer it to be a separate PR. For this PR, I recommend moving them into the reduction header as @jrhemstad suggested. Then in a separate PR we can discuss how best to consolidate them. Agree?

Reply:
Agree.

Reply:
It seems using enum class in cython is not fully supported yet.
I've tried several workaround examples, all won't work since the reduce function has the enum class input.
Can I drop enum class from this PL?

https://github.com/cython/cython/issues/1603
https://github.com/cython/cython/issues/2749

Reply:
Did someone ask you to convert this to an `enum class`? I think it's fine to just use an `enum` for now.

Reply:
That's Jake's suggestion. https://github.com/rapidsai/cudf/pull/1005#discussion_r273515907